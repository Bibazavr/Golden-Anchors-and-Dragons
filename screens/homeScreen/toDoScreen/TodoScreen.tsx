import * as React from "react";
import {FlatList, StyleSheet, TextInput, View} from "react-native";
import {TodoItem} from "./components/TodoItem";
import {onChangeTodoItemChecked} from "./utils/onChangeTodoItemChecked";
import {onDeleteTodoItem} from "./utils/onDeleteTodoItem";
import {Button} from "react-native-elements";
import {addTodoItem} from "./utils/addTodoItem";


export interface TodoScreenState {
    todoItem: { title: string, key: string, checked: boolean }
    list: TodoScreenState["todoItem"][]
}

export const TodoScreen = () => {
    const [list, setList] = React.useState<TodoScreenState["list"]>([])
    const [todoItem, setTodoItem] = React.useState<string>("");

    return <View style={stylesTodoScreen.container}>
        <TextInput style={stylesTodoScreen.input}
                   onChangeText={(inputText) => setTodoItem(inputText)}
                   placeholder={"Введите todo"}/>

        <Button style={stylesTodoScreen.buttonAdd} title={"add"}
                disabled={todoItem.length === 0}
                onPress={() => setList(addTodoItem(list, todoItem))}/>

        <FlatList data={list} renderItem={({item}) => <TodoItem name={item.title} key={item.key}
                                                                checked={item.checked}
                                                                onDeleteTodoItem={() => setList(onDeleteTodoItem(list, item.key))}
                                                                onChangeCheck={() => setList(onChangeTodoItemChecked(list, item.key))}/>}/>
    </View>
}

const stylesTodoScreen = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
    },
    input: {
        backgroundColor: "#f3f3f3",
        height: 30,
        margin: 10
    },
    buttonAdd: {
        margin: 10
    }
});
