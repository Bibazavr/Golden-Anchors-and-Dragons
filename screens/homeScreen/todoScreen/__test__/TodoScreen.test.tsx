import renderer from "react-test-renderer";
import * as React from "react";
import {TodoScreen} from "../TodoScreen";

it(`renders correctly`, () => {
    const tree = renderer.create(<TodoScreen/>).toJSON();


    expect(tree).toMatchSnapshot();
});