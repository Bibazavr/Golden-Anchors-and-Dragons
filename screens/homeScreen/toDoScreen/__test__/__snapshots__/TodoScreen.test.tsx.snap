// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Введите todo"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#f3f3f3",
        "height": 30,
        "margin": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 10,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(208, 8%, 90%)",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": Object {
                "color": Array [
                  208,
                  8,
                  63,
                ],
                "model": "hsl",
                "valpha": 1,
              },
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          add
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
